x/genutil/client/cli/validate_genesis.go

```diff
@@ -54,3 +51,19 @@ func ValidateGenesisCmd(mbm module.BasicManager, txEncCfg client.TxEncodingConfi
 		},
 	}
 }
+
+// validateGenDoc reads a genesis file and validates that it is a correct
+// Tendermint GenesisDoc. This function does not do any cosmos-related
+// validation.
+func validateGenDoc(importGenesisFile string) (*tmtypes.GenesisDoc, error) {
+	genDoc, err := tmtypes.GenesisDocFromFile(importGenesisFile)
+	if err != nil {
+		return nil, fmt.Errorf("%s. Make sure that"+
```

Comment:
It looks like this:
```
error reading GenesisDoc at /var/folders/rz/h235vrqd3lzgs09jxp_4jv540000gn/T/TestIntegrationTestSuite2_TestValidateGenesis_FromV037347171449/001/605276036: evidence.MaxAgeNumBlocks must be greater than 0. Got 0. Make sure that you have correctly migrated all Tendermint consensus params, please see the chain migration guide at https://docs.cosmos.network/master/migrations/chain-upgrade-guide-040.html for more info
```

Reply:
Does this validate a genesis file that is supposed to be in the format of 0.40? If so, I don't think this would error on the issue I opened. `max_bytes`, from my understanding, is allowed to be `0`. The genesis may be valid, but the intended behaviour of the genesis may not be what the entity migrating wanted

Reply:
I'm not sure I undestand. This function simply defers validating the genesis file to TM, btw max_bytes is not allowed to be 0, see ValidateConsensusParams in TM.

Reply:
Any valid genesis with 
```
 "evidence": {
      "max_age_num_blocks": "100000",
      "max_age_duration": "172800000000000",
      "max_bytes": "0"
    },
```
will pass the module manager validate genesis 

```
gaiad validate-genesis ~/genesis.json 
validating genesis file at /root/genesis.json
File at /root/genesis.json is a valid genesis file
```

This is because on the [SDK](https://github.com/cosmos/cosmos-sdk/blob/e4e39234c8e76626e5e9a36898daae1d7af78ff0/baseapp/params.go#L66) we check that max bytes isn't less than 0. If TM says 0 is an invalid value for max bytes, then so should the SDK. This should also reduce the likelyhood that an application doesn't set max bytes because then it would error by default

Reply:
I see that the error message on the linked function indicates that the max bytes must be positive despite only checking for negativity. I will open a pr

Reply:
nvm, [TM](https://github.com/tendermint/tendermint/blob/e3b96107af324fc6a7761563c4cf58d6c4119ad6/types/params.go#L112) only check that max bytes is non-negative

Reply:
> nvm, TM only check that max bytes is non-negative

oh true. Anyways, I added a test with max_bytes=0, and it's valid.

> This is because on the SDK we check that max bytes isn't less than 0. If TM says 0 is an invalid value for max bytes, then so should the SDK. This should also reduce the likelyhood that an application doesn't set max bytes because then it would error by default

Ideally, all the [params check the sdk does](https://github.com/cosmos/cosmos-sdk/blob/e4e39234c8e76626e5e9a36898daae1d7af78ff0/baseapp/params.go#L66) (that you linked) should call tendermint validation function under the hood. But that's maybe for another PR.

Reply:
> Ideally, all the params check the sdk does (that you linked) should call tendermint validation functions under the hood. But that's maybe for another PR.

I generally agree, but there is also an argument not to. If the SDK ever wants to not be dependent on having Tendermint as the plugged in consensus algo, then it is better that the consensus params validation remains in the SDK and just becomes more modular

Reply:
So, I think maybe we could just add a println after the error return that says to double check the consensus params? Maybe even specifically reference values whose default values are valid

Reply:
done!